tests/scripts/run-xunit-perf.py

```diff
@@ -0,0 +1,464 @@
+    myEnv = env
+    benchnameWithExt = benchname + '.' + testFileExt
+    fullPath = os.path.join(benchdir, benchnameWithExt)
+    shutil.copy2(fullPath, sandboxDir)
+
+    files = glob.iglob(os.path.join(benchdir, "*.txt"))
+    for filename in files:
+        if os.path.isfile(filename):
+            shutil.copy2(filename, sandboxDir)
+
+    myEnv['CORE_ROOT'] = sandboxDir
+
+    benchnameLogFileName = os.path.join(benchmarkOutputDir, lvRunId + '-' + benchname + '.log')
+
+    if not os.path.isdir(benchmarkOutputDir):
+        os.makedirs(benchmarkOutputDir)
+
+    log('')
+    log('--------')
+    log('Running %s %s' % (lvRunId, benchname))
+    log('--------')
+    log('')
+
+    lvCommonArgs = [os.path.join(sandboxDir, benchnameWithExt),
+            '--perf:outputdir',
+            benchmarkOutputDir,
+            '--perf:runid',
+            lvRunId]
+
+    if isScenarioTest:
+        lvCommonArgs += ['--target-architecture',
+                arch]
+    else:
+        lvCommonArgs.insert(0, 'PerfHarness.dll')
+
+    splitPrefix = [] if stabilityPrefix is None else stabilityPrefix.split(' ')
+    runArgs = executable + splitPrefix + [os.path.join(sandboxDir, 'corerun' + extension)] + lvCommonArgs + ['--perf:collect', collectionFlags]
+    log(" ".join(runArgs))
+    sys.stdout.flush()
+
+    error = 0
+    expectedOutputFile = os.path.join(benchmarkOutputDir, lvRunId + '-' + benchname + '.xml')
+    with open(benchnameLogFileName, 'wb') as out:
+        proc = subprocess.Popen(' '.join(runArgs), shell=True, stdout=out, stderr=out, env=myEnv)
+        proc.communicate()
+        if not proc.returncode == 0:
+            error = proc.returncode
+
+    if not error == 0:
+        log("CoreRun.exe exited with %s code" % (error))
+        if os.path.isfile(benchnameLogFileName):
+            with open(benchnameLogFileName, 'r') as f:
+                print(f.read())
+        return error
+    elif not os.path.isfile(expectedOutputFile):
+        log("CoreRun.exe failed to generate results in %s." % expectedOutputFile)
+        return 1
+
+    return 0
+
+def generate_results_for_benchview(python, lvRunId, benchname, isScenarioTest, better, hasWarmupRun, benchmarkOutputDir, benchviewPath):
+    benchviewMeasurementParser = 'xunitscenario' if isScenarioTest else 'xunit'
+    warmupRun = '--drop-first-value' if hasWarmupRun else ''
+    lvMeasurementArgs = [benchviewMeasurementParser,
+            '--better',
+            better,
+            warmupRun,
+            '--append']
+
+    filename = os.path.join(benchmarkOutputDir, lvRunId + '-' + benchname + '.xml')
+
+    runArgs = [python, os.path.join(benchviewPath, 'measurement.py')] + lvMeasurementArgs + [filename]
+    log('')
+    log(" ".join(runArgs))
+
+    proc = subprocess.Popen(runArgs, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```

Comment:
Prefer something like:

```python
try:
  subprocess.check_output(runArgs, stderr=PIPE)
except CalledProcessError as e:
  log(e.output.decode('utf-8'))
  raise RuntimeException("Call to measurement.py failed.")
```

Reply:
The functional reason I suggest this is that it will prevent splitting up stdout and stderr in your logs. Keeps context

Reply:
cool. didn't know about check_output